test-data/unit/check-redefine.test

```diff
@@ -102,7 +103,52 @@ def f() -> None:
     y
     y = ''
 
-def g(): pass
+[case testRedefineLocalWithinTryClauses]
```

Comment:
Could you check it's not possible to redefine a variable in a finally in a try? Like:

```py
try:
  try:
    ...
  finally:
    a = ""
    a = 5  # I don't think this should work
    fn_int(a)  # same as above
except:
  pass
```

Reply:
My patch still rejects this, consistent with previous treatment of `try` body. Within `try` we consider any statement to be fallible, so redefinitions are prohibited - this sounds reasonable to me. Added an explicit `testRedefineNestedInTry`. Do you think mypy should behave differently here?

Reply:
No, that sounds good. Just wanted to be sure this PR doesn't remove that treatment.